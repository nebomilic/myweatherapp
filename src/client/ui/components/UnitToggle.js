import React from 'react';
import PropTypes from 'prop-types';
const UnitToggle = ({clickHandler}) => { 
    return (
        <div className='toggle-container' >
            <div className='toggle' >  
                <span className='toggleText'>&deg;F</span>          
                <input type="checkbox" id="switch" onClick={clickHandler}/>
                <label htmlFor="switch">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;</label>
                <span className='toggleText'>&deg;C</span>
            </div>  
        </div>  

    ); 
};   

UnitToggle.propTypes = {
    clickHandler: PropTypes.func
};

export default UnitToggle;