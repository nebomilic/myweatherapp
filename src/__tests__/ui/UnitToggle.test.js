import React from 'react';
import UnitToggle from '../../client/ui/components/UnitToggle';
import renderer from 'react-test-renderer';

it ('renders corectly', () => {
    const clickHandler = () => alert('Toggle was clicked');
    const tree = renderer
        .create(<UnitToggle clickHandler={clickHandler}/>)
        .toJSON();

    expect(tree).toMatchSnapshot();
})