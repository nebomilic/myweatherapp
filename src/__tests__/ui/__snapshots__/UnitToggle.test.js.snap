// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders corectly 1`] = `
<div
  className="toggle-container"
>
  <div
    className="toggle"
  >
    <span
      className="toggleText"
    >
      °F
    </span>
    <input
      id="switch"
      onClick={[Function]}
      type="checkbox"
    />
    <label
      htmlFor="switch"
    >
                 
    </label>
    <span
      className="toggleText"
    >
      °C
    </span>
  </div>
</div>
`;
